platforms/native/engine/jsb-game.js

```diff
@@ -24,13 +24,18 @@
  ****************************************************************************/
 
 cc.game.restart = function () {
-    // Need to clear scene, or native object destructor won't be invoke.
-    cc.director.getScene().destroy();
-    cc.Object._deferredDestroy();
-    __restartVM();
+    return new Promise((resolve) => {
+        // Need to clear scene, or native object destructor won't be invoked.
+        cc.director.getScene().destroy();
+        cc.Object._deferredDestroy();
+        __restartVM();
+
+        // Resolve the promise to indicate completion
+        resolve();
+    });
```

Comment:
Doesn't need to put the __restartVM to Promise, just return `Promise.resolve();` is ok.

Reply:
Yep, i will test it.